rust/src/csv/reader.rs

```diff
@@ -78,16 +197,41 @@ impl Reader {
         let csv_reader = csv::ReaderBuilder::new()
             .has_headers(has_headers)
             .from_reader(BufReader::new(file));
-
         let record_iter = csv_reader.into_records();
         Reader {
-            schema: schema.clone(),
+            schema: schema,
             projection,
             record_iter,
             batch_size,
         }
     }
 
+    pub fn from_builder(file: File, builder: ReaderBuilder) -> Self {
```

Comment:
This should be changed to a method like `build` in `ReaderBuilder`.

Reply:
Done